### PR TITLE
SoliCurses: init at unstable-2022-02-13

### PR DESCRIPTION
**Pull request by laalsaas**
###### Description of changes

Add SoliCurses, an ncurses solitaire game, see: https://github.com/KaylaPP/SoliCurses

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/solicurses/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    repo = "SoliCurses";
+    rev = "dc89ca00fc1711dc449d0a594a4727af22fc35a0";
+    sha256 = "sha256-zWYXpvEnViT/8gsdMU9Ymi4Hw+nwkG6FT/3h5sNMCE4=";
+  };
+
+  buildInputs = [
+    ncurses
+  ];
+
+ # makefile = "build/Makefile";
```
Comment (AndersonTorres) [inline pkgs/games/solicurses/default.nix:22]:
```suggestion
```

File: pkgs/games/solicurses/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ preBuild = ''
+   cd build
+ '';
+  installPhase = ''
+    mkdir -p $out/bin
+    ls -R
+    install SoliCurses.out $out/bin/solicurses
+  '';
```
Comment (AndersonTorres) [inline pkgs/games/solicurses/default.nix:30]:
```suggestion

installPhase = ''
  install SoliCurses.out $out/bin/solicurses
'';
```

File: pkgs/games/solicurses/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  meta = with lib; {
+    description = "A version of Solitaire written in C++ using the ncurses library";
+    homepage = "https://github.com/KaylaPP/SoliCurses";
+    maintainers = with maintainers; [ laalsaas ];
+    license = licenses.gpl3Only;
```
Comment (AndersonTorres) [inline pkgs/games/solicurses/default.nix:36]:
```suggestion
    license = licenses.gpl3Only;
    inherit (ncurses.meta) platforms;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/games/solicurses/default.nix (hunk 2 above)
Comment (laalsaas) [reply pkgs/games/solicurses/default.nix:30]: admited, the `ls -R` was for debugging and isn't necessary, but without the `mkdir` the `install` fails with "no such file or directory"

Comment (AndersonTorres) [reply pkgs/games/solicurses/default.nix]: Sorry, I forgot the flag `-D`.

The command is `install SoliCurses.out -D $out/bin/solicurses`
 `

File: pkgs/games/solicurses/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  buildInputs = [
+    ncurses
+  ];
+
+ preBuild = ''
+   cd build
+ '';
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/games/solicurses/default.nix:25]: The indentation is wrong here. Fix this.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by laalsaas** [comment]
@AndersonTorres
Done, can you have yet another look at it?

File: pkgs/games/solicurses/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  buildInputs = [
+    ncurses
+  ];
+
+  preBuild = ''
+    cd build
+  '';
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/games/solicurses/default.nix:25]: Put a blank line here.

```suggestion

  installPhase = ''
```

File: pkgs/games/solicurses/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, ncurses
+}:
+
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/games/solicurses/default.nix:7]: Remove unneeded `rec`

```suggestion
stdenv.mkDerivation {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**